### PR TITLE
#4 Added options to manipulate border color, text color, and fill color

### PR DESCRIPTION
**Pull request by chrisahardie**
I removed all the cosmetic styles from the CSS and allowed for styles to be declared directly on elements via data-* attributes or programmatically via the options object we pass in.

### CODE REVIEW

File: js/percircle.js [hunk 1]
```diff
@@ -2,36 +2,66 @@
...
+              function(){
+                  span.css('color', textColor);
+              },
+              function(){
+                  span.css('color', '');
+              }
+            );
+
+            //Add child element to apply background color
+            var after = $('<div class="after"></div>');
```
Comment (chrisahardie) [inline js/percircle.js]: We can't (easily) manipulate pseudo-selectors, so I added a child div to simulate the same behaviour.

File: js/percircle.js [hunk 2]
```diff
@@ -2,36 +2,66 @@
...
+            perCircle.find('.bar, .fill').css('border-color', borderColor);
+
+            //Add text and apply text color
             var text = perCircle.attr('data-text') || options.text || percent + '%';
             $('<span>'+text+'</span>').appendTo(perCircle);
-            // add divs for structure
-            $('<div class="slice"><div class="bar"></div><div class="fill"></div></div>').appendTo(perCircle);
+            var span = perCircle.find('span');
+            var textColor = perCircle.data('text-color') || options.textColor;
+            perCircle.hover(
```
Comment (chrisahardie) [inline js/percircle.js]: `:hover` pseudoclass isn't (easily) manipulated in js, added a hover listener instead.

**Comment by toubou91** [comment]
Hey @chrisahardie ,

this seems a promising one, but we would better don't lose control of the initial behavior.
That is, we have to keep the on hover animation and also provide the two initial default styles (light/dark).

The second one, is actually translated to an implementation like the following one: the default style of a percircle object should be the lighter one, whereas if the user wants the dark one, this should be defined (i.e. data-style="dark"). So, at this point, there's no need for him/her to define to extra-options provided in this PR. These, must only be defined, when the user wants to define a fully custom styling for border, fill and text color.

Could you please handle the above two issues?

**Comment by chrisahardie** [comment]
Hey,
Sorry for the lack of response, I've been away. Works has been hectic so I haven't had a chance to look at this. Hopefully if I get some time this weekend I'll take a look at this.
Cheers,
C.

Date: Sat, 31 Oct 2015 16:36:27 -0700

**Comment by toubou91** [comment]
Hey @chrisahardie ,

same situation here, so, no worries for this one. 
So, what are your plans for this one?

Regards,
Thodoris

**Pull request closed without merging**